plugins/org.eclipse.sirius.diagram.sequence/src/org/eclipse/sirius/diagram/sequence/business/internal/layout/horizontal/SequenceHorizontalLayout.java

```diff
@@ -633,6 +636,51 @@ private boolean layoutReflexiveMessages(Map<? extends ISequenceElement, Rectangl
         return applied;
     }
 
+    private boolean layoutInteractionContainer(Map<? extends ISequenceElement, Rectangle> bounds, boolean pack) {
+        boolean applied = false;
+        Set<InteractionContainer> allInteractionContainers = this.sequenceDiagram.getAllInteractionContainers();
+        if (!allInteractionContainers.isEmpty() && getInteractionContainerBounds() != null) {
+            // Reset width of the interaction container
+            Bounds interactionContainerBounds = getInteractionContainerBounds();
```

Comment:
Use a Rectangle as bounds accumulator to avoid multiple setX/setHeight/width/y on the GMF Node layout constraints and all the notifications and so on